Source/CombatExtended/CombatExtended/ArmorUtilityCE.cs

```diff
@@ -127,6 +127,16 @@ public static DamageInfo GetAfterArmorDamage(DamageInfo originalDinfo, Pawn pawn
                         && app.def.apparel.CoversBodyPart(hitPart)
                         && !TryPenetrateArmor(dinfo.Def, app.PartialStat(dinfo.Def.armorCategory.armorRatingStat, hitPart), ref penAmount, ref dmgAmount, app))
                     {
+                        if (dinfo.Def.armorCategory.armorRatingStat == StatDefOf.ArmorRating_Sharp)
+                        {
+                            if (pawn.TryGetComp<Comp_BurnDamageCalc>() != null)

```

Comment:
`TryGetComp` is *quite* expensive.  Should cache the result, rather than calling it 3 times (I wouldn't rely on the JIT to optimize this).